manim/utils/testing/_test_class_makers.py

```diff
@@ -39,38 +45,49 @@ class _TestRenderer(from_renderer):
 class DummySceneFileWriter(SceneFileWriter):
     """Delegate of SceneFileWriter used to test the frames."""
 
-    def __init__(self, renderer, scene_name, **kwargs):
+    def __init__(
+        self,
+        renderer: CairoRenderer | OpenGLRenderer,
+        scene_name: StrPath,
+        **kwargs: Any,
+    ) -> None:
         super().__init__(renderer, scene_name, **kwargs)
         self.i = 0
 
-    def init_output_directories(self, scene_name):
+    def init_output_directories(self, scene_name: StrPath) -> None:
         pass
 
-    def add_partial_movie_file(self, hash_animation):
+    def add_partial_movie_file(self, hash_animation: str) -> None:
         pass
 
-    def begin_animation(self, allow_write=True):
+    def begin_animation(
+        self, allow_write: bool = True, file_path: StrPath | None = None
+    ) -> Any:
         pass
 
-    def end_animation(self, allow_write):
+    def end_animation(self, allow_write: bool = False) -> None:
         pass
 
-    def combine_to_movie(self):
+    def combine_to_movie(self) -> None:
         pass
 
-    def combine_to_section_videos(self):
+    def combine_to_section_videos(self) -> None:
         pass
 
-    def clean_cache(self):
+    def clean_cache(self) -> None:
         pass
 
-    def write_frame(self, frame_or_renderer, num_frames=1):
+    def write_frame(
+        self, frame_or_renderer: npt.NDArray | OpenGLRenderer, num_frames: int = 1
+    ) -> None:
```

Comment:
Could you please also update the parent class' type hint as well?
```suggestion
    def write_frame(
        self, frame_or_renderer: PixelArray | OpenGLRenderer, num_frames: int = 1
    ) -> None:
```